syntax = "proto3";

package discovery;

service LeaderDiscovery {
  rpc WhoIsLeader(Empty) returns (LeaderInfo);
  rpc UpdateLeader(LeaderInfo) returns (Empty);
}

message Empty {}

message LeaderInfo {
  string leader_id = 1;
  string host = 2;
  int32 port = 3;
}
